CenterForOpenScience/osf.io
# [Feature] Add preprint provider button to get share token/source [OSF-7997]

## Purpose

Adding tokens and sources via admin requires SHARE devs do things and then admins to update things. This PR makes this (theoretically) a one-click process!

## Changes
Add a button that gets the token/share source directly from SHARE to streamline things. This button will not function for local development (DEBUG_MODE=True) unless a `SHARE_PREPRINT_PROVIDER_PREPEND` is added to local.py (e.g., `SHARE_PREPRINT_PROVIDER_PREPEND = 'alexschiller'`. This namespaces local preprint provider data that is sent to SHARE, e.g., `alexschiller_agrixiv`

New button (With no share token/source):
<img width="723" alt="screen shot 2017-11-17 at 10 59 26 am" src="https://user-images.githubusercontent.com/1322421/32956320-705313b6-cb86-11e7-8794-2279bd8a7770.png">

New Disabled button (With share token/source already in place):
<img width="738" alt="screen shot 2017-11-17 at 10 59 02 am" src="https://user-images.githubusercontent.com/1322421/32956344-81039a0a-cb86-11e7-9e72-10e9db9bf47e.png">


Credit reel:
@caseyrollins for the WIP version of this in early July! (Been on hold for just a little while...)
@aaxelb for getting the SHARE api to play


## Side effects
If you have a share source or a share token already the button is disabled. Deleting both will make the button available again.


## Deployment configuration
Staging environments will need to set a SHARE_PREPRINT_PROVIDER_PREPEND that equals their staging environment name (staging, staging2, test). It is very important that test.osf.io is updated before this is released.


## Ticket

https://openscience.atlassian.net/browse/OSF-7997

## QA Notes
1. Test with a preprint provider that already has a share source/token. Note the token values, then delete them. Click the shiny new button and confirm that the share_source and access_token come back and match the previously noted ones

2. Make sure new providers get tokens too
